### PR TITLE
Additional data sets for maximum indexing throughput benchmarks

### PR DESCRIPTION
**Pull request by cdahlqvist**
I have created 2 new data sets, `eventdata` and `so`, that can be used for maximum indexing throughput benchmarks.

The `eventdata` track contains generated, enriched Apache access log records based on statistical distribution of a sample of access logs from the elastic.co website. The logs has been anonymized.

The `so` track contains questions and answers based on StackOverflow data. It contains a number of structured fields as well as a field with a significant text component.

### CODE REVIEW

File: eventdata/README.txt [hunk 1]
```diff
@@ -0,0 +1,10 @@
+EventData Data Set
+==================
+
+This data set contains 20 million Apache access log entries generated based on sample 
+elastic.co access logs using the generator avilable here: https://github.com/elastic/rally-eventdata-track
+
+The size of the data file is around 15GB, which gives an average JSON record size of 822 bytes. mappings have been somewhat optimized and the _all field has been disabled.
+
+The purpose of this track is to provide an efficient way to benchmark indexing of this data type as the generator
+built into the rally-eventdata-track can be CPU intensive.
```
Comment (danielmitterdorfer) [inline eventdata/README.txt]: As you are have generated the data could you please define a license explicitly here? Maybe https://creativecommons.org/licenses/by-sa/4.0/?

File: eventdata/challenges/default.json [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+          "warmup-time-period": 120,
+          "time-period": 3600,
+          "clients": 16
+        },
+        {
+          "operation": "force-merge",
+          "clients": 1
+        }
+      ]
+    }
```
Comment (danielmitterdorfer) [inline eventdata/challenges/default.json]: nit: missing new line

File: eventdata/mappings.json [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        "norms": false,
+        "type": "text",
+        "fields": {
+          "keyword": { "ignore_above": 256, "type": "keyword" }
+        }
+      }
+    },
+    "_all": { "enabled": false }
+  }
+}
```
Comment (danielmitterdorfer) [inline eventdata/mappings.json]: nit: missing new line

File: eventdata/mappings.json [hunk 4]
```diff
@@ -0,0 +1,43 @@
+{
+  "type": {
```
Comment (danielmitterdorfer) [inline eventdata/mappings.json]: maybe it makes sense to add `"dynamic": "strict"` here too?

File: eventdata/mappings.json [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    "properties": {
+      "@timestamp": { "type": "date" },
```
Comment (danielmitterdorfer) [inline eventdata/mappings.json]: do you intentionally avoid specifying the format?

File: eventdata/operations/default.json [hunk 6]
```diff
@@ -0,0 +1,9 @@
+    {
+      "name": "index-append",
+      "operation-type": "index",
+      "bulk-size": 5000
+    },
+    {
+      "name": "force-merge",
+      "operation-type": "force-merge"
+    }
```
Comment (danielmitterdorfer) [inline eventdata/operations/default.json]: nit: missing new line

File: eventdata/track.json [hunk 7]
```diff
@@ -0,0 +1,28 @@
+{% import "rally.helpers" as rally with context %}
+
+{
+  "short-description": "Eventdata indexing benchmark",
+  "description": "This benchmark indexes HTTP access logs generated based sample logs from the elastic.co website using the generator available in https://github.com/elastic/rally-eventdata-track",
+  "data-url": "http://benchmarks.elasticsearch.org.s3.amazonaws.com/corpora/eventdata",
```
Comment (danielmitterdorfer) [inline eventdata/track.json]: I checked the bucket and there is even a 1k version of the data. Thank you! :)

File: geonames/challenges/default.json [hunk 8]
```diff
@@ -192,6 +192,7 @@
           "clients": 1
         }
       ]
+<<<<<<< HEAD
```
Comment (danielmitterdorfer) [inline geonames/challenges/default.json]: this looks like an error during merge? I think you just need to remove this line, then it should be fine.

File: geonames/challenges/default.json [hunk 9]
```diff
@@ -213,3 +214,6 @@
         }
       ]
     }
+=======
+    }
+>>>>>>> b3185ba0a3d7fd798dfd55a35fbb10a12ebc5ffd
```
Comment (danielmitterdorfer) [inline geonames/challenges/default.json]: same as above (merge error)

File: so/README.txt [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+    
+	questionId:	      a unique ID for a question
+	answerId:         a unique ID for an answer
+	acceptedAnswerId: the unique ID of the answer accepted for question
+	title:	          a free-text field with the question title
+	creationDate:	  The date the questions was asked 
+	user:	          The user's unique ID
+	tags:	          An array of tags describing the technologies.
+    body:             Field contsaining the text of the question or answer.
+    type:             Type of post. Either 'question' or 'answer'
```
Comment (danielmitterdorfer) [inline so/README.txt]: nit: the indentation looks a bit broken.

File: so/challenges/default.json [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+          "warmup-time-period": 120,
+          "time-period": 3600,
+          "clients": 16
+        },
+        {
+          "operation": "force-merge",
+          "clients": 1
+        }
+      ]
+    }
```
Comment (danielmitterdorfer) [inline so/challenges/default.json]: nit: missing new line

File: so/track.json [hunk 12]
```diff
@@ -0,0 +1,28 @@
...
+{
+  "short-description": "Indexing benchmark using up to questions and answers from StackOverflow",
+  "description": "This test indexes up to 36,062,278 questions and answers from StackOverflow (total 33.12 GB json)",
+  "data-url": "http://benchmarks.elasticsearch.org.s3.amazonaws.com/corpora/so",
+  "indices": [
+    {
+      "name": "so",
+      "types": [
+        {
+          "name": "so",
```
Comment (danielmitterdorfer) [inline so/track.json]: the type in the mapping is called `question`.

File: so/mappings.json [hunk 13]
```diff
@@ -0,0 +1,37 @@
...
+			},
+			"type": {
+				"type": "keyword"
+			}
+		},
+		"_all": { 
+		    "enabled": false
+		}
+	}
+}
```
Comment (danielmitterdorfer) [inline so/mappings.json]: nit: missing new line. Also the indentation of the file looks a bit odd?

File: so/track.json [hunk 14]
```diff
@@ -0,0 +1,28 @@
+{% import "rally.helpers" as rally with context %}
+
+{
+  "short-description": "Indexing benchmark using up to questions and answers from StackOverflow",
```
Comment (danielmitterdorfer) [inline so/track.json]: ".. using up to .." <- there seems to be something missing.

**Comment by cdahlqvist** [comment]
@danielmitterdorfer Thanks for the feedback. Have updated and added License.

File: eventdata/README.txt [hunk 15]
```diff
@@ -8,3 +7,13 @@ The size of the data file is around 15GB, which gives an average JSON record siz
 
 The purpose of this track is to provide an efficient way to benchmark indexing of this data type as the generator
 built into the rally-eventdata-track can be CPU intensive.
+
+## License
+
+We use the same license for the data as the original data from Geonames:
```
Comment (danielmitterdorfer) [inline eventdata/README.txt]: copy & paste error (reference to Geonames)? :)

**Comment by cdahlqvist** [comment]
@danielmitterdorfer That was indeed a copy and paste error.... (blushing)

Have now corrected it.

**Comment by danielmitterdorfer** [comment]
Thanks for the corrections. Github just shows "This branch cannot be rebased due to conflicts" at the moment so I think you need to merge `origin/master` but then it should be good to go. 

You can do a quick verification if everything is fine syntax-wise by running `esrally list tracks` after merging. If it doesn't show any errors you should be good to merge the PR.

**Comment by danielmitterdorfer** [comment]
@cdahlqvist is there still interest in merging this PR? In general, I'm fine with you merging it after the conflicts are resolved. /cc: @dliappis

**Comment by danielmitterdorfer** [comment]
I'm closing this PR without merging for now due to inactivity. Please comment / reopen if there is interest in working on this again.

**Pull request closed without merging**